mitodl/ocw-studio
# Django 4.2.11 / Python 3.12.2 upgrade

### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-studio/issues/2140

### Description (What does it do?)
This PR upgrades Django to 4.2.11, along with some other dependencies that need to be upgraded to support it. Python was also upgraded to the latest supported version, 3.12.2.

### How can this be tested?
 - Run `docker compose up --build` on this branch to rebuild your docker containers with the latest versions of everything
 - Visit http://localhost:8043/ and make sure the site is reachable
 - If you are set up for publishing sites locally, test doing so
